### DISCUSSION TITLE
Couple of questions around the UI/Front End

### DISCUSSION THREAD

**Discussion by fkc1e100** [GoogleCloudPlatform/cluster-toolkit#920] [Q&A] [answered]
Hi,
Is there any documentation on the following:
1. Adding a second user account to the HPC Toolkit Front end?
2. Adding in different system shapes to the cluster template?  (Currently can only see c2-standard-60)  The documentation states that it is possible but guidance is light.
3. How to change a users password for the Front end?

Thanks!

**Comment by ikatsardis** [comment, marked as answer]
Hey Frank!

1. The OFE also comes with a admin panel that enables the administration tasks. This can be found at _https://YOUR_DEPLOYMENT_IP_OR_DNS/admin_ . There are currently 2 ways an admin can add a new user:
  Option 1 (Add the user directly): In this case the admin goes to _https://YOUR_DEPLOYMENT_IP_OR_DNS/admin/ghpcfe/user/_ and clicks the ADD USER icon, the next screen allows you to define a username and password. Once that is done anyone with those credentials can login to the OFE site. Admins can also go to the user's details and add the full name and email associated with it, along with any added permissions they want to give the user.
  Option 2 -Preferred- (Enable the user to register via the OAuth): In this case OAuth will handle the authentication of the users and there is no requirement for setting a password on the OFE site. To enable a user or a domain of users to register in the OFE the admin must add their emails in the _https://YOUR_DEPLOYMENT_IP_OR_DNS/admin/ghpcfe/authoriseduser/_ Once this is done, the respective user can simply login to the cluster via OAuth and utilizing the email account it was activated for them. Upon successful first login the account will be created, and the admin can navigate to the USERS page in the admin panel to edit their permissions. 
2. C2 is just a default setting, during the process of creating a new cluster the user is allowed to select from the dropdown menu any VM the associated project has access to. They can also create as many queues as they would like:
![Screenshot 2023-02-22 at 12 30 42 PM](https://user-images.githubusercontent.com/8696835/220620318-23952275-4352-4f9b-83ed-a747dd15c4e9.png)

3. If the process has been done via Option 1, simply click the user and change the password from the admin panel. If it was via Option 2 (OAuth) there is no password handling on our side and we rely on OAuth for that.

**Reply by fkc1e100** [reply]
Hi Ilias - thanks for the tips.  Questions 1 & 3 are good.

For Question 2 I'm not seeing any options in the drop down (see image below):
![image](https://user-images.githubusercontent.com/78167014/221382561-718dfc44-d7e9-49db-914a-afd087b64248.png)

Do I need to configure the compute nodes somewhere?

Thanks!
frank.

**Reply by ikatsardis** [reply]
Hey Frank,

The Front End will query gcloud and gather all the available machines that your project has access to. If you click the arrow it should be showing you all those and then you can select. In case it doesn't work, I would advice you check your permissions for the credential account you used, or any other organization policy. It could be a good idea also to have the front end auto create those credentials for you when you are deploying.
